frontend/dockerfile/docs/reference.md

```diff
@@ -320,6 +320,20 @@ directive in your Dockerfile:
   string=foobarbaz echo ${string%%b*}   # foo
   ```
 
+- `${variable/pattern/replacement}` replace the first occurrence of `pattern`
+  in `variable` with `replacement`
+
+  ```bash
+  string=foobarbaz echo ${string/ba/fo}  # fooforbaz
+  ```
+
+- `${variable//pattern/replacement}` replaces all occurrences of `pattern`
+  in `variable` with `replacement`
+
+  ```bash
+  string=foobarbaz echo ${string//ba/fo}  # fooforfoz
+  ```
```

Comment:
Could you add a note that this is only available in the master branch?

Reply:
David added the [notice](https://github.com/moby/buildkit/commit/4f459a4cd1cb3adc3e59278a4598f8cd58c31812#diff-fd3bfe0fa50e7d34c75c732a874a6909354fb5c13c70582d3fe8457f1547c075R291) "The following variable replacements are supported in a pre-release version of Dockerfile syntax, when using the `# syntax=docker/dockerfile-upstream:master` syntax directive in your Dockerfile:" in the previous PR.
There's a somewhat large list with experimental features above it, but in the rendered document the indentation should make it clear.